lib/matplotlib/_type1font.py

```diff
@@ -768,13 +861,262 @@ def transform(self, effects):
+        """Run the charstring interpreter on a glyph or subroutine.
+
+        This does not actually execute the code but simulates it to find out
+        which subroutines get called when executing the glyph or subroutine.
+
+        Returns
+        -------
+            glyphs : set[str]
+                The set of glyph names that are called by the glyph or subroutine
+            subrs : set[int]
+                The set of subroutines that are called by the glyph or subroutine
+            buildchar_stack : list[float]
+                The buildchar stack at the end of the simulation
+            postscript_stack : list[float]
+                The PostScript stack at the end of the simulation
+        """
+        if isinstance(glyph_or_subr, str):
+            program = self.prop['CharStrings'][glyph_or_subr]
+            glyphs = {glyph_or_subr}
+            subrs = set()
+        else:
+            program = self.prop['Subrs'][glyph_or_subr]
+            glyphs = set()
+            subrs = {glyph_or_subr}
+        for opcode in self._charstring_tokens(program):
+            if opcode in ('return', 'endchar'):
+                return glyphs, subrs, buildchar_stack, postscript_stack
+            newglyphs, newsubrs, buildchar_stack, postscript_stack = \
+                self._step(buildchar_stack, postscript_stack, opcode)
+            glyphs.update(newglyphs)
+            subrs.update(newsubrs)
+        else:
+            font_name = self.prop.get('FontName', '(unknown)')
+            _log.info(
+                f"Glyph or subr {glyph_or_subr} in font {font_name} does not end "
+                "with return or endchar"
+            )
+            return glyphs, subrs, buildchar_stack, postscript_stack
+
+    def _step(
+            self,
+            buildchar_stack: list[float],
+            postscript_stack: list[float],
+            opcode: int | str,
+        ) -> tuple[set, set, list[float], list[float]]:
+        """Run one step in the charstring interpreter."""
+        if isinstance(opcode, int):
+            return set(), set(), buildchar_stack + [opcode], postscript_stack
+        elif opcode in {
+                'hsbw', 'sbw', 'closepath', 'hlineto', 'hmoveto', 'hcurveto',
+                'hvcurveto', 'rlineto', 'rmoveto', 'rrcurveto', 'vhcurveto',
+                'vlineto', 'vmoveto', 'dotsection', 'hstem', 'hstem3', 'vstem',
+                'vstem3', 'setcurrentpoint'
+        }:
+            return set(), set(), [], postscript_stack
+        elif opcode == 'seac':
+            codes = buildchar_stack[3:5]
+            glyphs: set[str] = {self.prop['Encoding'][x] for x in codes}
+            return glyphs, set(), [], postscript_stack
+        elif opcode == 'div':
+            num1, num2 = buildchar_stack[-2:]
+            return (
+                set(),
+                set(),
+                buildchar_stack[-2:] + [num1/num2], postscript_stack
+            )
```

Comment:
This fits in a line?
```suggestion
            return set(), set(), buildchar_stack[-2:] + [num1/num2], postscript_stack
```